torch/distributed/rpc/api.py

```diff
@@ -294,6 +294,18 @@ def _init_rpc_backend(
     _set_and_start_rpc_agent(rpc_agent)
 
 
+def async_function(fn):
+    r"""
+    Wrapper of a function indicating that the return value of the function is
+    guranteed to be a ``torch.futures.Future``.
```

Comment:
nit: guaranteed

Can we add some detailed documentation here in terms of the purpose of this decorator and how it is supposed to be used? Also, we should probably mention this is a decorator instead of saying "Wrapper of a function" ?